include/stdalgos/detail/merge.hpp

```diff
@@ -0,0 +1,180 @@
+            // overload in std. Included only for completeness. This is not meant to be
+            // customized. It should fall back to the very default implementation.
+            template <typename InputIt1, typename InputIt2, typename OutputIt,
+                typename... Properties>
+            OutputIt operator()(
+                execution_properties<Properties...> const& exec_properties,
+                InputIt1 b1, InputIt1 e1, InputIt2 b2, InputIt2 e2,
+                OutputIt d) const
+            {
+                // Fall back to synchronizing the asynchronous overload if no synchronous
+                // customization is available.
+                // The sync_wait sender needs unwrapping since it returns an optional of tuple. And this
+                // overload returns an the output iterator.
+                return std::get<0>(stdexec::this_thread::sync_wait(
+                    merge_t{}(
+                        stdexec::just(b1, e1, b2, e2, d), exec_properties))
+                                       .value());
+            }
+
+            // TODO: add the overload that requires tag_invocable merge
+
+            // Default synchronous implementation with a scheduler but without a
+            // tag_invoke customization: This falls back to sync_waiting one of the
+            // asynchronous overloads.
+            template <stdexec::scheduler Scheduler, typename InputIt1,
+                typename InputIt2, typename OutputIt>
+            requires
+                // clang-format off
+      (!stdexec::tag_invocable<merge_t, Scheduler, InputIt1, InputIt1, InputIt2, InputIt2, OutputIt>)
+                // clang-format on
+                OutputIt
+                operator()(Scheduler&& sched, InputIt1 b1, InputIt2 e1,
+                    InputIt2 b2, InputIt1 e2, OutputIt d) const
+            {
+                // The sync_wait sender needs unwrapping since it returns an optional of tuple. And this
+                // overload returns an the output iterator.
+                return std::get<0>(stdexec::this_thread::sync_wait(
+                    merge_t{}(stdexec::transfer_just(
+                        std::forward<Scheduler>(sched), b1, e1, b2, e2, d)))
+                                       .value());
+            }
+
+            // Asynchronous overloads
+
+            // Use empty execution properties by default, forward to other overloads
+            template <stdexec::sender Sender>
+            stdexec::sender auto operator()(Sender&& sender) const
+            {
+                return merge_t{}(
+                    std::forward<Sender>(sender), execution_properties<>{});
+            }
+
+            // TODO: Add the overloads with tag_invocable and not tag_invocable with completion scheduler
+            template <typename InputIt1, typename InputIt2, typename OutputIt>
+            OutputIt merge_impl(InputIt1&& b1, InputIt1&& e1, InputIt2&& b2,
+                InputIt2&& e2, OutputIt&& d)
```

Comment:
You can take the iterators by value instead, i.e.
```suggestion
            OutputIt merge_impl(InputIt1 b1, InputIt1 e1, InputIt2 b2,
                InputIt2 e2, OutputIt d)
```